File: bank/src/main/java/com/bank/dto/CreditCardInfoDto.java
package com.bank.dto;

import jakarta.validation.constraints.Digits;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.CreditCardNumber;

import java.time.LocalDate;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
public class CreditCardInfoDto {

    @CreditCardNumber(message = "Credit card number is not valid")
    private String pan;

    @Digits(integer = 3, fraction = 0, message = "Invalid CVV")
    private String cvv;

    private String cardholderName;

    private LocalDate expirationDate;
}

File: bank-front/src/app/dto/CreditCardInfoDTO.ts
export class CreditCardInfoDTO {
    pan: string
    cvv: string
    cardholderName: string
    expirationDate: Date

    constructor({ pan, cvv, cardholderName, expirationDate }: any) {
        this.pan = pan;
        this.cvv = cvv;
        this.cardholderName = cardholderName;
        this.expirationDate = expirationDate;
      }
}
